wonnx/src/compiler.rs

```diff
@@ -771,7 +771,7 @@ pub fn compile(
         op @ ("MaxPool" | "AveragePool" | "Conv" | "ConvRelu" | "ConvLeakyRelu" | "ConvMish"
         | "GlobalAveragePool") => {
             // TODO: Conv only support NxCxHxW for the moment.
-            debug_assert!(input_shapes[0].rank() == 4);
+            assert!(input_shapes[0].rank() == 4);
```

Comment:
This one should not actually be an assert, instead an if check returning `Err(CompileError::InvalidInputShape{..})` when the condition is not met. The input ONNX model could simply be wrong and that should not cause a panic.